### PR TITLE
EAR 1072 QCodes on/off

### PR DESCRIPTION
**Pull request by farres1**
> BEFORE MAKING YOUR PR... There must be no linting errors, all tests must pass and PR is named after JIRA ticket number e.g. EAR-###.

### What is the context of this PR?

Implement a toggle switch allowing Authors to set if a questionnaire has QCodes

### How to review

- [ ] Check the toggle switch changes QCode nav item disabled state
- [ ] Check the error badge is removed from nav item when toggle switch is off for all QCode errors
- [ ] Check the error badge is displayed from nav item when toggle switch is on for all QCode errors
- [ ] Check the toggle switch updates qcodes value in Author schema
- [ ] Check the toggle switch updates data_verson in Runner schema

### CODE REVIEW

**Review by tjamesmac — CHANGES_REQUESTED** [review]
All looks great Sam, just need a test or two to support the qCode toggling on the `GeneralSettingsPage`

File: eq-author/src/App/QuestionnaireDesignPage/MainNavigation/index.js [hunk 1]
```diff
@@ -178,12 +179,17 @@ export const UnwrappedMainNavigation = ({
...
+                    totalErrorCount > 0 ||
+                    qcodesEnabled === false
+                  }
                 >
                   <IconText nav icon={qcodeIcon}>
                     QCodes
                   </IconText>
-                  {emptyQCode || duplicateQCode === true ? (
+                  {(qcodesEnabled && emptyQCode) ||
+                  (qcodesEnabled && duplicateQCode === true) ? (
```
Comment (tbaggaley) [inline eq-author/src/App/QuestionnaireDesignPage/MainNavigation/index.js:192]: Just a little one

```suggestion
                  {qcodesEnabled && (emptyQCode || duplicateQCode) ? (
```

**Review by tbaggaley — CHANGES_REQUESTED** [review]
Looking good Sam, works well.

Just one small suggestion.

Might be a good idea to add a migration so existing questionnaires have the toggle turned on - otherwise might confuse people when this goes live and they lose access to the qcodes page. I know it's not in the AC though

**Review by tbaggaley — DISMISSED** [review]

**Review by tjamesmac — CHANGES_REQUESTED** [review]
Hi Sam, sorry one more test please, could you assert that when `questionnaire.qCodes` is true or false, that the qCodes button in the main navigation is in fact disabled

**Review by tjamesmac — APPROVED** [review]
Nice one!

**Review by jasehumphr — APPROVED** [review]
Lovely Sam

**Pull request merged**